Source/OxyPlot/Series/BarSeries/RectangleBarSeries.cs

```diff
@@ -302,5 +324,75 @@ protected virtual bool IsValid(double v)
         {
             return !double.IsNaN(v) && !double.IsInfinity(v);
         }
-    }
+
+		/// <summary>
+		/// Find index of max(x) &lt;= target x in a list of data points
+		/// </summary>
+		/// <param name='items'>
+		/// vector of data points
+		/// </param>
+		/// <param name='targetX'>
+		/// target x.
+		/// </param>
+		/// <param name='guess'>
+		/// initial guess index.
+		/// </param>
+		/// <returns>
+		/// index of x with max(x) &lt;= target x or -1 if cannot find
+		/// </returns>
+		public static int FindBarIndex(IList<RectangleBarItem> items, double targetX, int guess)
```

Comment:
This looks like a duplicate of the `FindIndex` method below. I think you can create a generic version - with a `Func<TItem, double>` to give the X0.